pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -775,10 +786,28 @@ func getLocationFromPod(clientset kubernetes.Interface, d *apps.Deployment) (str
 			hostName = pvcName
 		}
 	}
-	return GetLocationWithNode(clientset, nodeName, hostName)
+	// compatibility with existing clusters
+	crushRoot := crushRootDefault
+	// find the env var indicating the root crush entry
+	for _, container := range pods.Items[0].Spec.Containers {
+		if container.Name != "osd" {
+			continue
+		}
+
+		for _, envvar := range container.Env {
+			if envvar.Name == "ROOK_CRUSHMAP_ROOT" {
```

Comment:
Please create a constant for `ROOK_CRUSHMAP_ROOT` since it's called multiple times.